ql/src/java/org/apache/hadoop/hive/ql/exec/util/Retryable.java

```diff
@@ -166,6 +169,25 @@ public synchronized Builder withRetryOnExceptionList(final List<Class<? extends
       return this;
     }
 
+    public synchronized Builder withFailOnParentException(final Class<? extends Exception> exceptionClass) {
+      if (exceptionClass != null &&
+              runnable.failOnParentExceptions.stream().noneMatch(k -> exceptionClass.equals(k))) {
+        runnable.failOnParentExceptions.add(exceptionClass);
+      }
+      return this;
+    }
+
+    public synchronized Builder withFailOnParentExceptionList(final List<Class<?
+            extends Exception>> exceptionClassList) {
+      for (final Class<? extends Exception> exceptionClass : exceptionClassList) {
+        if (exceptionClass != null &&
+                runnable.failOnParentExceptions.stream().noneMatch(k -> exceptionClass.equals(k))) {
+          runnable.failOnParentExceptions.add(exceptionClass);
+        }
```

Comment:
Rather than checking for a unique entry ourselves, can we not use a set instead of a list?

Reply:
we are using the same data structure to maintain retryOnExceptionList and failOnExceptionList also.